src/ring/middleware/authorization.clj

```diff
@@ -0,0 +1,58 @@
+(ns ring.middleware.authorization
+  (:require [clojure.string :as str]
+            [ring.util.parsing :as parsing]))
+
+(def ^:private re-token68
+  (re-pattern "[A-Za-z0-9._~+/-]+=*"))
+
+(def ^:private re-auth-param
+  (re-pattern (str "(" parsing/re-token ")\\s*=\\s*(?:" parsing/re-value ")")))
+
+(defn- parse-auth-params
+  [auth-params]
+  (reduce (fn [m kv]
+            (if-let [[_ k v1 v2] (re-matches re-auth-param kv)]
+              (assoc m (str/lower-case k) (or v1 v2))
+              m))
+          {}
+          (str/split auth-params #"\s*,\s*")))
+
+(defn- parse-authorization
+  [request]
+  (when-let [[auth-scheme token-or-params]
+             (some-> (get-in request [:headers "authorization"])
+                     (str/split #"\s" 2))]
+    (cond
+      (empty? token-or-params)
+      {:scheme (str/lower-case auth-scheme)}
+
+      (re-matches re-token68 token-or-params)
+      {:scheme  (str/lower-case auth-scheme)
+       :token68 token-or-params}
+
+      :else
+      {:scheme (str/lower-case auth-scheme)
+       :params (parse-auth-params token-or-params)})))
+
+(defn- authorization-request
```

Comment:
This should be public. It's useful for systems that don't use middleware directly, such as Pedestal.